src/CatenaLogic.Analyzers/Analyzers/Diagnostics/Base/DiagnosticRuleBase.cs

```diff
@@ -2,7 +2,7 @@
 {
     using Microsoft.CodeAnalysis.Diagnostics;
 
-    public abstract class AnalyzerBase : IAnalyzer
+    public abstract class DiagnosticRuleBase : IDiagnostic
```

Comment:
IDiagnostic vs DiagnosticRule? Is this the recommended naming style according to official examples?